plugins/discovery/endpoint_creator_test.go

```diff
@@ -0,0 +1,73 @@
+package discovery
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/wavefronthq/wavefront-collector-for-kubernetes/internal/discovery"
+	"github.com/wavefronthq/wavefront-collector-for-kubernetes/internal/metrics"
+	"github.com/wavefronthq/wavefront-collector-for-kubernetes/internal/util"
+	"github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/discovery/prometheus"
+	"github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/discovery/telegraf"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func makeDummyProviders(handler metrics.ProviderHandler) map[string]discovery.ProviderInfo {
+	providers := make(map[string]discovery.ProviderInfo, 2)
+	providers["prometheus"] = prometheus.NewProviderInfo(handler, "prom.")
+	providers["telegraf"] = telegraf.NewProviderInfo(handler)
+	return providers
+}
+
+func Test_endpointCreator_discoverEndpoints_annotations_dont_happen_when_disabled(t *testing.T) {
+	e := &endpointCreator{
+		delegates:                  nil,
+		providers:                  makeDummyProviders(util.NewDummyProviderHandler(1)),
+		disableAnnotationDiscovery: true,
+	}
+
+	resource := discovery.Resource{
+		Kind:       discovery.PodType.String(),
+		IP:         "0.0.0.0",
+		Meta:       metav1.ObjectMeta{},
+		Containers: make([]v1.Container, 0),
+	}
+
+	got := e.discoverEndpoints(resource)
+	assert.Equal(t, 0, len(got))
+}
+
+func Test_endpointCreator_discoverEndpoints_annotations_happen_when_not_disabled(t *testing.T) {
+	e := &endpointCreator{
+		delegates:                  nil,
+		providers:                  makeDummyProviders(util.NewDummyProviderHandler(1)),
+		disableAnnotationDiscovery: false,
+	}
+
+	resource := discovery.Resource{
+		Kind:       discovery.PodType.String(),
+		IP:         "0.0.0.0",
+		Meta:       metav1.ObjectMeta{},
+		Containers: make([]v1.Container, 0),
+	}
+
+	got := e.discoverEndpoints(resource)
```

Comment:
Curious why this matches when the resource does not have any annotations on it (under the `meta` section)?

Reply:
It works because down in the encoder it assumes that we're using an annotation and then fills in all the default values when nothing is defined in meta:
`func Param(meta metav1.ObjectMeta, annotation, cfgVal, defaultVal string) string {
	value := ""
	// give precedence to annotation
	if annotation != "" {
		value = meta.GetAnnotations()[annotation]
	}
	if value == "" {
		// then config
		value = cfgVal
	}
	if value == "" {
		// then default
		value = defaultVal
	}
	return value
}`